esp-hal/src/dma/mod.rs

```diff
@@ -1518,18 +1553,28 @@ pub trait DmaChannel: crate::private::Sealed {
 pub trait Rx: crate::private::Sealed {
     fn init(&mut self, burst_mode: bool, priority: DmaPriority);
 
+    /// TODO: In the future, this will be removed in favour of
+    /// [Self::prepare_and_start];
     unsafe fn prepare_transfer_without_start(
         &mut self,
         peri: DmaPeripheral,
         chain: &DescriptorChain,
     ) -> Result<(), DmaError>;
 
+    /// TODO: In the future, this will be removed in favour of
+    /// [Self::prepare_and_start];
     unsafe fn prepare_transfer<BUF: DmaRxBuffer>(
         &mut self,
         peri: DmaPeripheral,
         buffer: &mut BUF,
     ) -> Result<(), DmaError>;
 
+    fn prepare_and_start<BUF: DmaRxBuffer>(
+        &mut self,
+        peri: DmaPeripheral,
+        buffer: BUF,
```

Comment:
Just keep taking BUF by reference and don't create the View so close to the DMA.

Reply:
Doing it this way means `prepare_and_start` can be a safe function. In future I'm also hoping for a `DmaTransfer` struct that owns the buffer/view and peripheral drivers shouldn't care too much what the DMA driver is doing.

Reply:
Let's revisit those ideas later.